### PR TITLE
Adding video model, 'create' mutation, and query 

### PR DESCRIPTION
**Pull request by kayladavis**

### CODE REVIEW

**Comment by billdybas** [comment]
I quickly glanced at the video regex, and I don't think it covers all cases. 

You can find more robust regexes and some examples here:
- YouTube:
https://github.com/CookPete/react-player/blob/master/src/players/YouTube.js#L8
https://github.com/CookPete/react-player/blob/master/test/specs/canPlay.js#L14-L18 (These probably also need more query parameter tests (eg. when `?v=` is `&v=`))
- Vimeo: 
https://github.com/CookPete/react-player/blob/master/src/players/Vimeo.js#L7
https://github.com/CookPete/react-player/blob/master/test/specs/canPlay.js#L43-L50

File: backend/constants.js [hunk 1]
```diff
@@ -4,3 +4,4 @@ export const JUDGE = 'JUDGE'
 export const IMAGE_ENTRY = 0x1
 export const VIDEO_ENTRY = 0x2
 export const OTHER_ENTRY = 0x3
+export const ALLOWED_VIDEO_PROVIDERS = new Set(['youtube', 'vimeo'])
```
Comment (robmcl4) [inline backend/constants.js]: Is there an immutable set?

File: backend/schema.js [hunk 2]
```diff
@@ -79,7 +79,6 @@ interface Entry {
     yearLevel: String
     academicProgram: String
     moreCopies: Boolean
-    excludeFromJudging: Boolean
```
Comment (robmcl4) [inline backend/schema.js]: ty

File: backend/constants.js (hunk 1 above)
Comment (kayladavis) [reply backend/constants.js]: apparently you can't do this w/o a dependency so it's staying as is.

**Pull request merged**